### PR TITLE
refactor(aws): Simplified ELB security group autocreation

### PR DESCRIPTION
**Pull request by jeyrschabu**
- updated code that allows ingress from elb group to app security group

### CODE REVIEW

**Comment by jeyrschabu** [comment]
@ajordens PTAL (cc @cfieber)

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy [hunk 1]
```diff
@@ -192,62 +191,88 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
...
+    }
 
-    IngressSecurityGroupDescription applicationLoadBalancerSecurityGroupDescription = new IngressSecurityGroupDescription(
-      applicationLoadBalancerSecurityGroupId,
-      applicationLoadBalancerSecurityGroupName,
-      description.credentialAccount,
-      description.vpcId
-    )
+    // include health check endpoint
+    Matcher portMatcher = HEALTHCHECK_PORT_PATTERN.matcher(description.healthCheck)
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy]: Is `description.healthCheck` the right attribute to check?

```
        "healthCheck": "HTTP:7001/healthcheck",
        "healthCheckProtocol": "HTTP",
        "healthCheckPort": 7001,
```

Could we just check `healthCheckPort`? Alternatively if we're checking `.healthCheck`, is there any implication if the value was `HTTP:7001/v2/healthcheck`?

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerClassicAtomicOperationSpec.groovy [hunk 2]
```diff
@@ -86,21 +86,57 @@ class UpsertAmazonLoadBalancerClassicAtomicOperationSpec extends Specification {
   def regionScopedProviderFactory = Stub(RegionScopedProviderFactory) {
     forRegion(_, "us-east-1") >> regionScopedProvider
   }
+
+  def securityGroupLookup = Mock(SecurityGroupLookup)
```
Comment (ajordens) [inline clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerClassicAtomicOperationSpec.groovy]: Out of curiosity, is there a test when the application security group does not exist? Looking at the code we'll create it right?

**Comment by ajordens** [comment]
Couple minor comments but the fix to how we're determining application names looks sensible to me (caveat Frigga vs. Moniker).

@lwander Any moniker-related concerns here? Not sure what the overarching goal is around people potentially creating load balancers that don't fit the existing naming patterns (fair bit of fallout if-so I suspect).

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy (hunk 1 above)
Comment (jeyrschabu) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy]: For simplicity, I will use `healthCheckPort` here

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerClassicAtomicOperationSpec.groovy (hunk 2 above)
Comment (jeyrschabu) [reply clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerClassicAtomicOperationSpec.groovy]: Yes. we will create it. I will add a test for that as well

**Comment by jeyrschabu** [comment]
Updated

**Pull request merged**